clients/src/test/java/org/apache/kafka/clients/consumer/internals/ConsumerCoordinatorTest.java

```diff
@@ -1299,6 +1377,178 @@ public void testForceMetadataDeleteForPatternSubscriptionDuringRebalance() {
         }
     }
 
+    @Test
+    public void testOnJoinPrepareWithOffsetCommitShouldSuccessAfterRetry() {
+        rebalanceConfig = buildRebalanceConfig(Optional.of("group-id"));
+        ConsumerCoordinator coordinator = buildCoordinator(rebalanceConfig,
+                new Metrics(),
+                assignors,
+                true,
+                subscriptions);
+        client.prepareResponse(groupCoordinatorResponse(node, Errors.NONE));
+        coordinator.ensureCoordinatorReady(time.timer(Long.MAX_VALUE));
+        subscriptions.subscribe(singleton(topic1), rebalanceListener);
+        client.prepareResponse(joinGroupFollowerResponse(1, consumerId, "leader", Errors.NONE));
+        client.prepareResponse(syncGroupResponse(singletonList(t1p), Errors.NONE));
+        coordinator.joinGroupIfNeeded(time.timer(Long.MAX_VALUE));
```

Comment:
Why can't we call `prepareCoordinatorForCloseTest` here? I think the only difference is we don't `coordinator.poll`, right? If so, I think we can add a parameter into `prepareCoordinatorForCloseTest` to determine if we need to poll or not, otherwise, there are many duplicated codes here. Ex:
```java
private ConsumerCoordinator prepareCoordinatorForCloseTest(final boolean useGroupManagement,
                                                               final boolean autoCommit,
                                                               final Optional<String> groupInstanceId,
                                                               final shouldPoll) {

...
if (shouldPoll) {
  coordinator.poll(time.timer(Long.MAX_VALUE));
}
}
```

WDYT?

Same comments applies to below new added tests.

Reply:
Also, since we created a new coordinator, we should close it at the end of test. That's If you agree with my above suggestion, we can do like this to close it:
```java
try (ConsumerCoordinator coordinator = prepareCoordinatorForCloseTest(...)) {


}
```

Reply:
Thanks! the test looks much simpler